### PR TITLE
Ap/zend 276

### PR DESCRIPTION
**Pull request by a-petrini**
This pull request follows https://github.com/bitcoin/bitcoin/pull/7692 and removes the p2p alert system.

### CODE REVIEW

File: qa/rpc-tests/blockdelay_2.py [hunk 1]
```diff
@@ -19,14 +19,9 @@
 import time
 class blockdelay_2(BitcoinTestFramework):
 
-    alert_filename = None
```
Comment (JackPiri) [inline qa/rpc-tests/blockdelay_2.py:22]: This alert file code can be removed also from following Python modules: `blockdelay.py` (not used at all, being in comment section) and `forknotify.py` (passed to nodes as startup parameter, but actually never used).

Also, `alert_filename = None` can be removed from `p2p_ignore_spent_tx.py` Python module.

File: src/main.cpp [hunk 2]
```diff
@@ -2160,6 +2158,24 @@ bool fLargeWorkForkFound = false;
 bool fLargeWorkInvalidChainFound = false;
 CBlockIndex *pindexBestForkTip = NULL, *pindexBestForkBase = NULL;
 
+void alertNotify(const std::string& strMessage, bool fThread) {
```
Comment (JackPiri) [inline src/main.cpp:2161]: I strongly suggest to name this function as `internalAlertNotify` or even `internalAlarmNotify` (or something similar) in order to make explicit this is not related to old alert system (messages received from P2P network) but rather to internal notification logic.

File: src/init.cpp [hunk 3]
```diff
@@ -342,7 +340,6 @@ std::string HelpMessage(HelpMessageMode mode)
 
     string strUsage = HelpMessageGroup(_("Options:"));
     strUsage += HelpMessageOpt("-?", _("This help message"));
-    strUsage += HelpMessageOpt("-alerts", strprintf(_("Receive and display P2P network alerts (default: %u)"), DEFAULT_ALERTS));
     strUsage += HelpMessageOpt("-alertnotify=<cmd>", _("Execute command when a relevant alert is received or we see a really long fork (%s in cmd is replaced by message)"));
```
Comment (JackPiri) [inline src/init.cpp]: Similarly to comment related to function `alertNotify`, I'd rephrase the description of `-alertnotify` parameter to something like `"Execute command when a relevant internal alert is raised (%s in cmd is replaced by message)"` (or something similar).

File: src/test/alert_tests.cpp [hunk 4]
```diff
@@ -3,385 +3,19 @@
...
-    BOOST_CHECK(!alerts[1].AppliesTo(999002, "/MagicBean:0.1.0/"));
-    BOOST_CHECK(!alerts[1].AppliesTo(1, "/MagicBean:0.2.0/"));
-
-    BOOST_CHECK(!alerts[2].AppliesTo(1, "/MagicBean:0.3.0/"));
-
-    SetMockTime(0);
-}
-
-
-BOOST_AUTO_TEST_CASE(AlertNotify)
```
Comment (JackPiri) [inline src/test/alert_tests.cpp:324]: Since we are keeping a portion of this file, should the test `AlertNotify` be kept (adapting it to use `alertNotify` method)?

File: src/Makefile.test.include [hunk 5]
```diff
@@ -36,7 +36,7 @@ JSON_TEST_FILES = \
   test/data/g1_compressed.json \
   test/data/g2_compressed.json
 
-#RAW_TEST_FILES = test/data/alertTests.raw
```
Comment (JackPiri) [inline src/Makefile.test.include:39]: It seems to me variables `RAW_TEST_FILES` and `GENERATED_TEST_FILES` can be dropped, correct?

**Review by JackPiri — APPROVED** [review]
The PR is fine to me.
I left a few comments for cleaning-up some unused/leftover code, minor importance.
Additionally, please have a look at this comment https://github.com/HorizenOfficial/zen/pull/540/files#r1182726188: I think some renaming should be performed (wherever the word _alert_ is mentioned) in order to make totally clear old alert system has definitely gone for good.
Thanks

File: src/main.cpp [hunk 6]
```diff
@@ -2160,6 +2158,24 @@ bool fLargeWorkForkFound = false;
...
+    // Alert text should be plain ascii coming from a trusted source, but to
+    // be safe we first strip anything not in safeChars, then add single quotes around
+    // the whole string before passing it to the shell:
+    std::string singleQuote("'");
+    std::string safeStatus = SanitizeString(strMessage);
+    safeStatus = singleQuote+safeStatus+singleQuote;
+    boost::replace_all(strCmd, "%s", safeStatus);
+
+    if (fThread)
+        boost::thread t(runCommand, strCmd); // thread runs free
```
Comment (drgora) [inline src/main.cpp]: I'd suggest replacing these `boost` (also at line 2171) functions with `std` alternatives, considering it should be low effort.

**Review by drgora — APPROVED** [review]
I left one non-blocking comment, but the PR is good by me.

File: qa/rpc-tests/blockdelay_2.py (hunk 1 above)
Comment (a-petrini) [reply qa/rpc-tests/blockdelay_2.py:22]: Thanks, removed from `block_delay.py` and `p2p_ignore_spent_tx.py`. Not sure about `forknotify.py`, but I agree that there are no explicit tests on the contents of `alert.txt` file, once it has been opened and - possibly - written to.

File: src/Makefile.test.include (hunk 5 above)
Comment (a-petrini) [reply src/Makefile.test.include:39]: Yes, but only `RAW_TEST_FILES`. Wanted to follow a comment on Bitcoin PR, but I do not see any purpose for keeping this one here.

File: src/init.cpp (hunk 3 above)
Comment (a-petrini) [reply src/init.cpp]: Agree, the help string has been rephrased.

File: src/main.cpp (hunk 2 above)
Comment (a-petrini) [reply src/main.cpp:2161]: While I agree that this function is triggered by an internal condition, rather than from a broadcasted message, I would keep the name as it is, to match Bitcoin code.

File: src/main.cpp (hunk 6 above)
Comment (a-petrini) [reply src/main.cpp]: Yes, I modified the function to accommodate to `std::regex_replace` instead of `boost::replace_all`, and `std::thread` instead of `boost::thread`.

File: src/test/alert_tests.cpp (hunk 4 above)
Comment (a-petrini) [reply src/test/alert_tests.cpp:324]: Yes, I added a Boost test that checks the function proper behavior.

File: src/main.cpp (hunk 6 above)
Comment (drgora) [reply src/main.cpp]: Great, thanks!

**Review by ptagl — APPROVED** [review]
